syntax = "proto3";

// Package for server and client must be identical! Otherwise connection won't be made and the UNIMPLEMENTED error will be raised.
package edu.gla.kail.ad.service;

option java_package = "edu.gla.kail.ad.service";
option java_multiple_files = true;

import "client.proto";

service AgentDialogue {
    // Send the response from agents using when InteractionRequest is received.
    rpc GetResponseFromAgents (InteractionRequest) returns (InteractionResponse) {
    }

    // Provide the streaming to send responses back to the UI.
    rpc ListResponses(InteractionRequest) returns (stream InteractionResponse) {
    }

    // End current session for a particular user.
    rpc EndSession (UserID) returns (UserID) {
    }
}

message UserID {
    string user_id = 1;
    bool activeSession = 2; // True - the user has an active session; false - the user doesn't have an active session.
}